### PR TITLE
Re-arrange WARNING messages about reference inconsistencies

### PR DESCRIPTION
**Pull request by cocoatomo**
Subject: Display appropriate WARNING messages about reference inconsistencies

### Feature or Bugfix

- Bugfix

### Purpose

- Separate correctly causes for each WARNING message. (Which message is about which cause?)

### Detail

For reference inconsistencies, now, Sphinx outputs four messages:

1. auto-numbered footnote: "inconsistent footnote references in translated message"
2. reference with refname: "inconsistent references in translated message"
3. reference: "inconsistent term references in translated message"
4. named footnote & citation: "inconsistent references in translated message"

(these messages are all implemented in sphinx/transforms/i18n.py.)

However, reST specification do not define "named footnote", which is actually citation.
Furthermore, citation inconsistencies are reported from warn_missing_reference method of ReferencesResolver class (which is located in sphinx/transforms/post_transforms/__init__.py) and
check_consistency function of StandardDomain class (which is located in sphinx/domains/std.py).

In conclusion, Sphinx should output four messages:

1. footnote: "inconsistent footnote references in translated message"
2. reference with refname: "inconsistent references in translated message"
3. reference: "inconsistent term references in translated message"
4. citation: "Citation [hoge] is not referenced." and "citation not found: fuga"

### Relates
- This PR is a preparation for #4106 .

### CODE REVIEW

File: sphinx/transforms/i18n.py [hunk 1]
```diff
@@ -258,20 +258,19 @@ def is_named_target(node):
                     (nodes.paragraph,) + LITERAL_TYPE_NODES + IMAGE_TYPE_NODES):
                 continue  # skip for now
 
-            # auto-numbered foot note reference should use original 'ids'.
```
Comment (tk0miya) [inline sphinx/transforms/i18n.py:261]: This block is only for auto-numbered footnote_refs. So please don't merge standard footnote_refs here.

File: sphinx/transforms/i18n.py [hunk 2]
```diff
@@ -327,26 +326,6 @@ def is_refnamed_ref(node):
 
                 self.document.note_refname(new)
 
-            # refnamed footnote and citation should use original 'ids'.
```
Comment (tk0miya) [inline sphinx/transforms/i18n.py:330]: To change the warning message, please make a copy of this block to separate verifications of foonote_refs and citation_refs.

**Review by tk0miya — CHANGES_REQUESTED** [review]
In addition, I guess new testcase is needed to check new warning message of non-auto-numbered footnote refs.

**Comment by tk0miya** [comment]
Ah, I'd forgotten to say you. This is a bugfix, so please rebase this to stable branch. Then it will released as next 1.6.x release.
Thank you for your help :-)

**Comment by cocoatomo** [comment]
I have re-created a revised PR #4124, because `git rebase` was involving unrelated file diffs.
